File: Java/RecursionPractice/src/calculations/CountDown.java
package calculations;

public class CountDown {

    int countDownFrom;

    public void output() {
        if (this.countDownFrom == 0) {
            System.out.println("Blast Off!");
            return;
        }
        System.out.println("The number is: " + this.countDownFrom);
        this.countDownFrom--;
        this.output();
    }

    public CountDown(int countDownFrom) {
        this.countDownFrom = countDownFrom;
    }
}

File: Kotlin/RecursionPractice/src/calculations/Countdown.kt
package calculations

class Countdown(private var countDownFrom: Int) {

    fun printCountdown() {
        if (countDownFrom == 0) {
            println("Blast Off!")
            return
        }
        println("The number is: $countDownFrom")
        countDownFrom--
        this.printCountdown()
    }
}